### ISSUE TITLE
unsafe_wrap cause dead kernel on 0.3.2? 

### ISSUE THREAD

**Issue by testercwt** [JuliaStrings/StringEncodings.jl#40]
In 0.3.2  , unsafe_wrap used on julia version 1.4.1 
```julia
@static if VERSION >= v"1.6.0-DEV.438"
        inbuf_view = view(s.inbuf, Int(s.inbytesleft[]+1):BUFSIZE)
    else
        inbuf_view = unsafe_wrap(Array, pointer(s.inbuf, s.inbytesleft[]+1), BUFSIZE)
    end
```
however, Bounderror or dead kernel occurred while decoding files ,  but encoding/decoding string is fine 
I wonder "unsafe_wrap" maybe the cause

**Comment by nalimilan** [comment]
Thanks for the report. Could you post code to reproduce this? Could you try replacing `if VERSION >= v"1.6.0-DEV.438"` with `if true` and see if the problem goes away?

**Comment by nalimilan** [comment]
Actually I think I've found the problem. I wonder why the tests didn't catch it, maybe because the test strings are too short. Please confirm whether version 0.3.3 fixes the crash.

**Comment by testercwt** [comment]
> Actually I think I've found the problem. I wonder why the tests didn't catch it, maybe because the test strings are too short. Please confirm whether version 0.3.3 ([JuliaRegistries/General#19227](https://github.com/JuliaRegistries/General/pull/19227)) fixes the crash.

it fixed on version 0.3.3

**Comment by testercwt** [comment]
> Thanks for the report. Could you post code to reproduce this? Could you try replacing `if VERSION >= v"1.6.0-DEV.438"` with `if true` and see if the problem goes away?

this also fix the problem
